src/lightly_train/_data/file_helpers.py

```diff
@@ -29,63 +30,65 @@
     PathLike,
 )
 
+logger = logging.getLogger(__name__)
+
 
 class ImageMode(Enum):
     RGB = "RGB"
     UNCHANGED = "UNCHANGED"
     MASK = "MASK"
 
 
-def list_image_files(imgs_and_dirs: Sequence[Path]) -> Iterable[Path]:
+def list_image_filenames_from_iterable(
+    imgs_and_dirs: Iterable[PathLike],
+) -> Iterable[ImageFilename]:
     """List image files recursively from the given list of image files and directories.
 
+    Assumes that all given paths exist.
+
     Args:
         imgs_and_dirs: A list of (relative or absolute) paths to image files and
             directories that should be scanned for images.
 
     Returns:
-        A list of absolute paths pointing to the image files.
+        An iterable of image filenames starting from the given paths. The given paths
+        are always included in the output filenames.
     """
+    supported_extensions = _pil_supported_image_extensions()
     for img_or_dir in imgs_and_dirs:
-        if img_or_dir.is_file() and (
-            img_or_dir.suffix.lower() in _pil_supported_image_extensions()
-        ):
-            yield img_or_dir.resolve()
-        elif img_or_dir.is_dir():
-            yield from _get_image_filepaths(img_or_dir)
+        _, ext = os.path.splitext(img_or_dir)
+        # Only check image extension. This is faster than checking isfile() because it
+        # does not require a system call.
+        if ext.lower() in supported_extensions:
+            yield ImageFilename(img_or_dir)
+        # For dirs we have to make a system call.
+        elif os.path.isdir(img_or_dir):
+            is_empty = True
+            dir_str = str(img_or_dir)
+            for image_filename in _get_image_filenames(image_dir=dir_str):
+                is_empty = False
+                yield ImageFilename(os.path.join(dir_str, image_filename))
+            if is_empty:
+                logger.warning(f"The data directory '{img_or_dir}' is empty.")
```

Comment:
This raised an error previously. But if you pass multiple images/dirs you don't really care if one of them is empty. There might be many reasons for it.